explainaboard/analysis/analyses.py

```diff
@@ -321,6 +333,253 @@ def perform(
+        sample_limit: an upper limit on the number of samples saved in each bucket.
+        cls_name: the name of the class.
+    """
+
+    feature: str
+    number: int = 10
+    sample_limit: int = 50
+    cls_name: Optional[str] = None
+
+    def __post_init__(self):
+        """Set the class name."""
+        self.cls_name: str = self.__class__.__name__
+
+    AnalysisCaseType = TypeVar('AnalysisCaseType')
+
+    def perform_calibration_analysis(
+        self, bucket_performances: list[BucketPerformance] = []
+    ) -> tuple[float, float]:
+        """Calculate the metrics for calibration analysis.
+
+        This includes Expected Calibration Error and Maximum Calibration Error.
+        """
+        total_error, total_size = 0.0, 0
+        MCE = 0.0
+        for bucket_performance in bucket_performances:
+            metric_result = bucket_performance.performances.get(
+                "Accuracy", Performance(0.0)
+            )
+            bucket_accuracy = metric_result.value
+            bucket_confidence = (
+                metric_result.auxiliary_result.confidence
+                if isinstance(metric_result.auxiliary_result, ConfidenceMetricResult)
+                else 0.0
+            )
+            bucket_size = bucket_performance.n_samples
+            total_error += bucket_size * abs(bucket_accuracy - bucket_confidence)
+            total_size += bucket_size
+            MCE = max(MCE, abs(bucket_accuracy - bucket_confidence))
+        return (total_error / total_size, MCE)
+
+    def perform(
+        self,
+        cases: list[AnalysisCase],
+        metrics: dict[str, Metric],
+        stats: dict[str, MetricStats],
+        confidence_alpha: float,
+    ) -> AnalysisResult:
+        """See Analysis.perform."""
+        if len(cases) == 0 or self.feature not in cases[0].features:
+            raise RuntimeError(
+                f"calibration analysis: feature {self.feature} not found."
+            )
+
+        acc_metric = metrics.get('Accuracy', None)
+        metric_stat = stats.get('Accuracy', None)
+        if not acc_metric or not metric_stat:
+            raise RuntimeError("calibration analysis: metric Accuracy not found.")
+
+        # Add confidence values to metric_stats
+        acc_data = metric_stat.get_data()
+        conf_data = np.expand_dims(
+            np.array([float(case.features.get(self.feature, 0.0)) for case in cases]), 1
+        )
+        assert acc_data.shape == conf_data.shape
+        metric_stat = SimpleMetricStats(np.concatenate([acc_data, conf_data], axis=-1))
+
+        # Preparation for bucketing
+        bucket_func: Callable[..., list[AnalysisCaseCollection]] = getattr(
+            explainaboard.analysis.bucketing,
+            "fixed",
+        )
+        bucket_size = 1.0 / self.number
+        bucket_setting = [
+            (
+                (i * bucket_size, (i + 1) * bucket_size)
+                if i < self.number - 1
+                else (i * bucket_size, 1.0)
```

Comment:
I think multiplication with a reciprocal causes certain amount of floating point errors.
Directly dividing the value would be better (and it is not so expensive in this case):
```suggestion
                (i / self.number, (i + 1) / self.number)
                if i < self.number - 1
                else (i / self.number, 1.0)
```